src/diffusers/quantizers/gguf/gguf_quantizer.py

```diff
@@ -0,0 +1,132 @@
+    is_accelerate_available,
+    is_accelerate_version,
+    is_gguf_available,
+    is_gguf_version,
+    is_torch_available,
+    logging,
+)
+
+
+if is_torch_available() and is_gguf_available():
+    import gguf
+    import torch
+
+    from .utils import (
+        GGUFParameter,
+        _quant_shape_from_byte_shape,
+        _replace_with_gguf_linear,
+    )
+
+
+logger = logging.get_logger(__name__)
+
+
+class GGUFQuantizer(DiffusersQuantizer):
+    def __init__(self, quantization_config, **kwargs):
+        super().__init__(quantization_config, **kwargs)
+
+        self.compute_dtype = quantization_config.compute_dtype
+        self.pre_quantized = True
+
+    def validate_environment(self, *args, **kwargs):
+        if not is_accelerate_available() or is_accelerate_version("<", "0.26.0"):
+            raise ImportError(
+                "Loading GGUF Parameters requires `accelerate` installed in your enviroment: `pip install 'accelerate>=0.26.0'`"
+            )
+        if not is_gguf_available() or is_gguf_version("<", "0.10.0"):
+            raise ImportError(
+                "To load GGUF format files you must have `gguf` installed in your environment: `pip install gguf>=0.10.0`"
+            )
+
+    def adjust_max_memory(self, max_memory: Dict[str, Union[int, str]]) -> Dict[str, Union[int, str]]:
+        # need more space for buffers that are created during quantization
+        max_memory = {key: val * 0.90 for key, val in max_memory.items()}
+        return max_memory
+
+    def adjust_target_dtype(self, target_dtype: "torch.dtype") -> "torch.dtype":
+        if target_dtype != torch.uint8:
+            logger.info(f"target_dtype {target_dtype} is replaced by `torch.uint8` for GGUF quantization")
+        return torch.uint8
+
+    def update_torch_dtype(self, torch_dtype: "torch.dtype") -> "torch.dtype":
+        if torch_dtype is None:
+            torch_dtype = self.compute_dtype
+        return torch_dtype
+
+    def check_quantized_param_shape(self, param_name, current_param, loaded_param):
+        loaded_param_shape = loaded_param.shape
+        current_param_shape = current_param.shape
+        quant_type = loaded_param.quant_type
+
+        block_size, type_size = gguf.GGML_QUANT_SIZES[quant_type]
+
+        inferred_shape = _quant_shape_from_byte_shape(loaded_param_shape, type_size, block_size)
+        if inferred_shape != current_param_shape:
+            raise ValueError(
+                f"{param_name} has an expected quantized shape of: {inferred_shape}, but receieved shape: {loaded_param_shape}"
+            )
+
+        return True
+
+    def check_if_quantized_param(
+        self,
+        model: "ModelMixin",
+        param_value: Union["GGUFParameter", "torch.Tensor"],
+        param_name: str,
+        state_dict: Dict[str, Any],
+        **kwargs,
+    ) -> bool:
+        if isinstance(param_value, GGUFParameter):
+            return True
+
+        return False
+
+    def create_quantized_param(
```

Comment:
For bitsandbytes, the bias is often left untouched and is kept as the original `dtype` (`torch.bfloat16`) for example. Does the same apply to GGUF or we can quantize everything?

Reply:
The default logic in llama.cpp is [here](https://github.com/ggerganov/llama.cpp/blob/59f4db10883a4f3e855cffbf2c3ab68430e95272/src/llama.cpp#L18922-L18926), the weights quantized with the comfy repo follow this logic, so biases and 1d tensors are left unquantized.

They're FP32 internally in llama.cpp since they're fairly small (and the operation for numpy/gguf FP32 -> torch FP32 doesn't require any dequantization, so doing that might be ever so slightly faster. Same goes for FP16, but *not* BF16 due to not being a native numpy datatype).

Reply:
Checking with @DN6 if we're following the same.

Reply:
Biases and 1D tensors are unquantized. @city96 created the GGUF quants for Flux and SD3.* so they're the expert here :)

Reply:
Thanks! I don't remember any specific code / tests for checking that.

Reply:
Ah seems like this:
https://github.com/huggingface/diffusers/pull/9964/files#diff-1aab1b9f58b9a73465bb88dbd921eeee13cca2913f9cb5808d14c3ca88f5b1b8R389

Maybe having a test for this would be nice!